contrib/storage-hive/core/src/test/java/org/apache/drill/exec/hive/complex_types/TestHiveStructs.java

```diff
@@ -234,6 +240,28 @@ public void primitiveStruct() throws Exception {
         .go();
   }
 
+  @Test // DRILL-7429
+  public void testCorrectColumnOrdering() throws Exception {
+    String sql = "SELECT t.str_n0 a, rid b FROM hive.struct_tbl t LIMIT 1";
+    DirectRowSet directRowSet = queryBuilder().sql(sql).rowSet();
+    BatchSchema fields;
+    try {
+      fields = directRowSet.batchSchema();
+    } finally {
+      directRowSet.clear();
+    }
+
+    assertThat(fields.getFieldCount(), is(2));
```

Comment:
We also have `RowSetComparison` class, which allows comparing expected and actual row sets, including their schema.

Or if you don't want to populate row set with expected data, you may use the `schemaBaseLine()` method from `TestBuilder` to validate query schema only.